### PR TITLE
Project Mongo API - Linda

### PR DESCRIPTION
**Pull request by lindahz**
Link to my deployed project:
https://lindas-project-mongo-api.herokuapp.com/

My reflections on how this week's project turned out:
I had limited time this week, my focus was to achieve only blue level requirements.

Things I'd like to have clarified or explained in more detail:
Nothing I can think of.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,35 +3,100 @@ import bodyParser from 'body-parser'
...
+  const singleID = await Character.findOne({ id: id })
+  if (singleID) {
+    res.json(singleID)
+  } else {
+    res.status(404).json(ERROR_CHARACTERS_NOT_FOUND)
+  }
+})
+
+// Shows all characters from a specific house, not case sensitive
+app.get('/characters/:house', async (req, res) => {
```
Comment (amandaflod) [inline server.js:90]: You could change the path to /characters/house/:house for better consistency, as you did so in the other path!

File: server.js [hunk 2]
```diff
@@ -3,35 +3,100 @@ import bodyParser from 'body-parser'
...
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+import harryData from './data/harry-potter-characters.json'
+
+const ERROR_CHARACTERS_NOT_FOUND = {error : 'No character results were found, please try again.'}
```
Comment (amandaflod) [inline server.js:8]: Good idea to create this variable!

File: server.js [hunk 3]
```diff
@@ -3,35 +3,100 @@ import bodyParser from 'body-parser'
...
+    res.json(singleID)
+  } else {
+    res.status(404).json(ERROR_CHARACTERS_NOT_FOUND)
+  }
+})
+
+// Shows all characters from a specific house, not case sensitive
+app.get('/characters/:house', async (req, res) => {
+  const { house } = req.params
+  const houses = await Character.find({ house: { $regex: house, $options: "i"} })
```
Comment (amandaflod) [inline server.js:92]: Great use of regexp to create a simple search!

**Review by amandaflod — COMMENTED** [review]
As always Linda - great job! Despite your time constraints that week you managed to build a well-working API and write smart and structured code! 😊 There really isn't much to comment on, I think you did good ❤️ (and sorry for late review!!!)

File: server.js [hunk 4]
```diff
@@ -3,35 +3,100 @@ import bodyParser from 'body-parser'
...
+  const characters = await Character.find(req.query) 
+  if (characters.length === 0) {
+    res.status(404).json(ERROR_CHARACTERS_NOT_FOUND)
+  } else {
+    res.json(characters)
+  }
+})
+
+// Shows one character with unique id
+app.get('/characters/id/:id', async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:79]: `id` is a default key so you should use endpoint `characters/:id`

File: server.js (hunk 1 above)
Comment (KarolCislowski) [reply server.js:90]: and here `house` is not the identification key for this collection so you should use `/characters/houses/:house`

**Review by KarolCislowski — APPROVED** [review]
Great job here 🥇

**Pull request closed without merging**